File: StkUiPlugins/CSharp/GroundVehicleFromDirectionsREST12/CustomConfigPage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using AGI.Ui.Plugins;
using AGI.STKObjects;

namespace Agi.Ui.Directions
{
    public partial class CustomConfigPage : UserControl, IAgUiPluginConfigurationPageActions2
    {
        public CustomConfigPage()
        {
            InitializeComponent();
        }

        private void checkBox1_CheckedChanged(object sender, EventArgs e)
        {
            //Enable apply button
            if (_site != null)
                _site.SetModified(true);
        }

        #region Members

            IAgUiPluginConfigurationPageSite _site;
            BasicCSharpPlugin _plugin;

        #endregion

            #region IAgUiPluginConfigurationPageActions2 Members

            public bool OnApply()
            {
                _plugin.Integrate = checkBox1.Checked;
                return true;
            }

            public void OnCancel()
            {
                // Intentionally left empty
            }

            public void OnCreated(IAgUiPluginConfigurationPageSite Site)
            {
                _site = Site;
                _plugin = _site.Plugin as BasicCSharpPlugin;
                checkBox1.Checked = _plugin.Integrate;
            }

            public void OnOK()
            {
                _plugin.Integrate = checkBox1.Checked;
            }

            public void OnHelp()
            {
                IAgUiPluginSite2 site2 = (_site.Plugin as BasicCSharpPlugin).Site as IAgUiPluginSite2;
                site2.ShowHelp( @"C:\Program Files\AGI\STK 9\Help\stk.chm", "", AgEUiPluginHelpDisplayType.eUiPluginHelpTOC);
            }
            #endregion
    }
}

File: StkUiPlugins/VB/ReentryHeatingPlugin/Stk12.ReentryHeatingPlugin/CustomConfigPage.vb
Imports Agi.Ui.Plugins
Imports Agi.STKObjects
Imports AGI.STKUtil


Public Class CustomConfigPage
    Implements IAgUiPluginConfigurationPageActions

    Private _site As IAgUiPluginConfigurationPageSite
    Private _plugin As Agi.Ui.Plugins.VB_Net.ReentryHeating.ReentryHeatingPlugin

    Public Function OnApply() As Boolean Implements Agi.Ui.Plugins.IAgUiPluginConfigurationPageActions.OnApply
        _plugin.Integrate = CheckBox1.Checked
        Return True
    End Function

    Public Sub OnCancel() Implements AGI.Ui.Plugins.IAgUiPluginConfigurationPageActions.OnCancel
        'Intentionally left empty
    End Sub

    Public Sub OnCreated(ByVal Site As AGI.Ui.Plugins.IAgUiPluginConfigurationPageSite) Implements AGI.Ui.Plugins.IAgUiPluginConfigurationPageActions.OnCreated
        _site = Site
        _plugin = DirectCast(_site.Plugin, ReentryHeatingPlugin)
        CheckBox1.Checked = _plugin.Integrate
    End Sub

    Public Sub OnOK() Implements AGI.Ui.Plugins.IAgUiPluginConfigurationPageActions.OnOK
        _plugin.Integrate = CheckBox1.Checked
    End Sub

    Private Sub CheckBox1_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CheckBox1.CheckedChanged
        'Enable apply button
        If Not (_site Is Nothing) Then
            _site.SetModified(True)
        End If
    End Sub

    Private Sub btnHelpDocument_Click(sender As System.Object, e As System.EventArgs) Handles btnHelpDocument.Click
        'open the Help Document
        Dim root As AgStkObjectRoot
        root = _plugin.STKRoot

        Dim stkInstallDirectory
        stkInstallDirectory = root.ExecuteCommand("GetDirectory / STKHome")


        Dim helpFileDirectory As String
        helpFileDirectory = CStr(stkInstallDirectory(0)) + "Plugins\ReentryHeatingPlugin\Reentry_Heating_Plugin_README.pdf"


        Try

            System.Diagnostics.Process.Start(helpFileDirectory)

        Catch

        End Try


    End Sub
End Class
